src/app/user/application/usecases/update-user-avatar.usecase.ts

```diff
@@ -0,0 +1,45 @@
+import { BadRequestException, Inject, InternalServerErrorException } from '@nestjs/common';
+import { IUserRepository } from '@app/user/domain/ports/iuser.repository';
+import { User } from '@app/user/domain/entities/user';
+
+
+
+interface FileData {
+  filename: string;
+}
+
+export class UpdateUserAvatarUsecase {
+  constructor(@Inject('UsersRepository') private readonly userRepository: IUserRepository) {}
+
+  async execute(user: User, file: FileData): Promise<{ avatar: string }> {
+    
+    if (!file || !file.filename) {
+      throw new BadRequestException('Avatar file is required.');
+    }
+
+    try {
+    
+      const baseUrl = process.env.APP_URL || 'http://localhost:3002'; 
+      const avatarUrl = `${baseUrl}/uploads/avatars/${file.filename}`;
```

Comment:
```suggestion
      const avatarUrl = `${baseUrl}/files/images/user_avatars/${file.filename}`;
```
اعمل فايل للconsts اللي زي هيك